qiskit/result/exceptions.py

```diff
@@ -39,3 +39,8 @@ def __init__(self, error):
 
     def __str__(self):
         return '{}: {}'.format(self.code, self.message)
+
+
+class NoMostFrequentCount(QiskitError):
+    """Error raised by Counts.most_frequent if there is no most frequent result."""
+    pass
```

Comment:
this is such a specific error (occurs in one method of one class), why does it need a separate exception class?

Reply:
I'll change it to a QiskitError directly instead of a subclass

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/4501/commits/a4e5ce59dd54b388666408a3a7da1b96cdebb59f